### PR TITLE
Add fuzzer in Xcrd package

### PR DESCRIPTION
**Pull request by AdamKorcz**
Signed-off-by: AdamKorcz <adam@adalogics.com>

### Description of your changes

Moves over two fuzzers from cncf-fuzzing: https://github.com/cncf/cncf-fuzzing/blob/main/projects/crossplane/xcrd_fuzzer.go

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review (but get the error `make: *** No rule to make target 'reviewable'. Stop.`)
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested
Both fuzzers can be tested with:
```
go test -fuzz=FuzzForCompositeResourceXcrd
go test -fuzz=FuzzForCompositeResourceClaim
```

from the `internal/xcrd` dir

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by AdamKorcz** [comment]
@muvaf @jbw976 Is there anything besides the conflict needs adjusting?

**Comment by jbw976** [comment]
@ulucinar would you be able to take a look at this contribution from @AdamKorcz, since you had been focusing recently on fuzz testing results/debugging? Thank you!! 🙇

**Review by phisco — APPROVED** [review]
LGTM, once this gets merged we can proceed moving here all the other tests too.

**Review by turkenh — APPROVED** [review]
LGTM, @AdamKorcz could you rebase your PR, there seem to be some conflicts.

**Pull request merged**
